test/src/keyboard_listener_test.cpp

```diff
@@ -0,0 +1,92 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "game_constants.h"
+#include "missile_launcher.h"
+#include "mocks/mock_keyboard_listener.h"
+#include "keyboard.h"
+#include "events.h"
+
+#include <SDL2/SDL.h>
+#include <tuple>
+
+using ::testing::_;
+using ::testing::Property;
+using ::testing::InSequence;
+
+class KeyboardListenerTest : public ::testing::TestWithParam<std::tuple<std::string , int>> {
+public: 
+    MockKeyboardListener *keyboard_listener;
+    Keyboard             *keyboard;
+    Events               *events;
+    SDL_Event            *sdlevent;
```

Comment:
Inconsistent naming here? Also I feel like the whitespacing, not everyone has a standard tool to make it this way like in go, let's just keep it easy and separate each word by one space. Also I feel like these should be private and have a trailing underscore.